desci-contracts/test/DpidAliasRegistry.ts

```diff
@@ -0,0 +1,227 @@
+    let proxyAdmin: string;
+
+    before(async () => {
+      reciept = await dpidAliasRegistry.deployTransaction.wait();
+      proxyAddress = reciept.contractAddress;
+      implAddress = await upgrades.erc1967.getImplementationAddress(proxyAddress);
+      proxyAdmin = await upgrades.erc1967.getAdminAddress(proxyAddress);
+
+      console.log({
+        implAddress,
+        proxyAdmin,
+        implOwner: await dpidAliasRegistry.owner(),
+      })
+    });
+
+    it("costs a reasonable amount of gas", async () => {
+      expect(
+        BigNumber.from(reciept.cumulativeGasUsed).lte(32000000),
+        `Gas limit exceeded`
+      ).to.be.true;
+    });
+
+    it("is made through proxy", () => {
+      expect(proxyAddress).not.to.equal(implAddress);
+    });
+
+    it("deploys implementation with proxy admin as owner", async () => {
+      const registryOwner = await dpidAliasRegistry.owner();
+      expect(registryOwner).to.equal(proxyAdmin);
+    });
+
+    it("respects initializer dpid offset", async () => {
+      const nextDpid = await dpidAliasRegistry.nextDpid();
+      expect(nextDpid).to.equal(FIRST_DPID);
+    })
+  });
+
+  describe("alias registry", () => {
+    const STREAM_A = "kjzl6kcym7w8y7i5ugaq9a3vlm7hhuaf4bpl5o5qykeh4qtsa12c6rb5ekw6aaa";
+    
+    describe("entry", () => {
+      let tx: ContractTransaction;
+      let res: ContractReceipt;
+      
+      before(async () => {
+        tx = await dpidAliasRegistry.mintDpid(STREAM_A);
+        res = await tx.wait();
+      });
+
+      it("can be added", async () => {
+        const entry = await dpidAliasRegistry.registry(100);
+        expect(entry).to.equal(STREAM_A);
+      });
+
+      it("emits event on mint", async () => {
+        const event = res.events![0];
+        const [ dpid, streamId ] = event.args!;
+
+        expect(event.event).to.equal("DpidMinted");
+        expect(dpid).to.equal(BigNumber.from(100));
+        expect(streamId).to.equal(STREAM_A);
+      });
+
+      it("increases counter on mint", async () => {
+        const nextDpid = await dpidAliasRegistry.nextDpid();
+        expect(nextDpid).to.equal(101);
+      });
+
+      it("gets next free dpid", async () => {
+        const STREAM_B = "kjzl6kcym7w8y7i5ugaq9a3vlm7hhuaf4bpl5o5qykeh4qtsa12c6rb5ekw6bbb";
+        const tx2 = await dpidAliasRegistry.mintDpid(STREAM_B);
+        await tx2.wait();
+        const entry = await dpidAliasRegistry.registry(101);
+        expect(entry).to.equal(STREAM_B);
+      });
+    });
+
+    describe("legacy entry", () => {
+      let migrationEntry: DpidAliasRegistry.LegacyDpidEntryStruct;
+        
+      before(async () => {
+        migrationEntry = {
+          owner: user1.address, // of dpid owner
+          versions: [
+            {
+              cid: "bafybeihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku",
+              timestamp: 1716369952,
+            },
+          ],
+        };
+      });
+      
+      describe("import", () => {
```

Comment:
Can you add a test to compare the gas cost of importing 150 legacy dPIDs during deployment and adding them individually like this?

Reply:
Had a chat with Sina and concluded that regardless of a difference in cost, it's smart to lean toward contract simplicity, so it'll be kept as a single import function